### PR TITLE
fix: log outputs

### LINKED ISSUE #311: logs for removed images
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
Users should be able to get a log output for the images removed by eraser pods.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Eraser version:
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by ashnamehrotra**
**What this PR does / why we need it**:
Cleans eraser code and log outputs for eraser to include digest and name of image.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #311

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ashnamehrotra** [comment]
I put some log outputs for collector, and they don't show if they are above or directly below the if statement on line 39 - all other lines show the log statements, so I'm guessing its due to the go routine. We don't have any log statements there right now, but I'm not sure if there is a fix for it in the case we do add log output there in the future

File: pkg/eraser/helpers.go [hunk 1]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
 			err = c.deleteImage(backgroundContext, digest)
 			if err != nil {
-				log.Error(err, "Error removing", "image", digest)
+				log.Error(err, "Error removing", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
 				continue
 			}
 
 			deletedImages[imgDigestOrTag] = struct{}{}
-			log.Info("Removed", "given", imgDigestOrTag, "digest", digest, "digest", digest)
+			log.Info("Removed", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
			log.Info("removed image", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```

File: pkg/eraser/helpers.go [hunk 2]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
 			if ex := util.IsExcluded(excluded, imgDigestOrTag, idToTagListMap); ex {
-				log.Info("Image is excluded", "image", imgDigestOrTag)
+				log.Info("Image is excluded", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
 				continue
 			}
 
 			err = c.deleteImage(backgroundContext, digest)
 			if err != nil {
-				log.Error(err, "Error removing", "image", digest)
+				log.Error(err, "Error removing", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
				log.Error(err, "error removing image", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```

File: pkg/eraser/helpers.go [hunk 3]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
 
 		if digest, isNonRunning := nonRunningImages[imgDigestOrTag]; isNonRunning {
 			if ex := util.IsExcluded(excluded, imgDigestOrTag, idToTagListMap); ex {
-				log.Info("Image is excluded", "image", imgDigestOrTag)
+				log.Info("Image is excluded", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
				log.Info("image is excluded", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```

File: pkg/eraser/helpers.go [hunk 4]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
-			log.Info("Removed", "given", imgDigestOrTag, "digest", digest, "digest", digest)
+			log.Info("Removed", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
 			continue
 		}
 
-		_, isRunning := runningImages[imgDigestOrTag]
+		digest, isRunning := runningImages[imgDigestOrTag]
 		if isRunning {
-			log.Info("Image is running", "image", imgDigestOrTag)
+			log.Info("Image is running", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
			log.Info("image is running", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
```

File: pkg/eraser/helpers.go [hunk 5]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
-			if _, running := runningImages[img]; running {
+			if util.IsExcluded(excluded, digest, idToTagListMap) {
+				log.Info("Image is excluded", "digest", digest, "name", idToTagListMap[digest])
 				continue
 			}
 
-			if util.IsExcluded(excluded, img, idToTagListMap) {
-				log.Info("Image is excluded", "image", img)
+			if err := c.deleteImage(backgroundContext, digest); err != nil {
+				log.Error(err, "Error during prune", "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
				log.Error(err, "error during prune", "digest", digest, "name", idToTagListMap[digest])
```

File: pkg/eraser/helpers.go [hunk 6]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
-		for img := range nonRunningImages {
-			if _, deleted := deletedImages[img]; deleted {
+		for _, digest := range nonRunningImages {
+			if _, deleted := deletedImages[digest]; deleted {
 				continue
 			}
 
-			if _, running := runningImages[img]; running {
+			if util.IsExcluded(excluded, digest, idToTagListMap) {
+				log.Info("Image is excluded", "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
				log.Info("image is excluded", "digest", digest, "name", idToTagListMap[digest])
```

File: pkg/eraser/helpers.go [hunk 7]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
-		_, isRunning := runningImages[imgDigestOrTag]
+		digest, isRunning := runningImages[imgDigestOrTag]
 		if isRunning {
-			log.Info("Image is running", "image", imgDigestOrTag)
+			log.Info("Image is running", "given", imgDigestOrTag, "digest", digest, "name", idToTagListMap[digest])
 			continue
 		}
 
-		log.Info("Image is not on node", "image", imgDigestOrTag)
+		log.Info("Image is not on node", "given", imgDigestOrTag)
```
Comment (sozercan) [inline pkg/eraser/helpers.go]:
```suggestion
		log.Info("image is not on node", "given", imgDigestOrTag)
```

File: pkg/eraser/helpers.go [hunk 8]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
 				continue
 			}
-
-			if err := c.deleteImage(backgroundContext, img); err != nil {
-				log.Error(err, "Error during prune", "image", img)
-				continue
-			}
-			log.Info("Prune successful", "image", img)
+			log.Info("Prune successful", "digest", digest, "name", idToTagListMap[digest])
+			deletedImages[digest] = struct{}{}
```
Comment (sozercan) [inline pkg/eraser/helpers.go:99]: what is this for?

Comment (ashnamehrotra) [reply pkg/eraser/helpers.go:99]: nonRunningImages is a map of (digest | tag) -> digest, so there is a chance the same digest is present twice when we iterate the values of the map. We added a check to skip if we had already deleted an image digest on line 83, but we were missing deletedImages[digest] = struct{}{} to update the map to hold the deleted values.

File: pkg/eraser/helpers.go [hunk 9]
```diff
@@ -54,50 +54,47 @@ func removeImages(c Client, targetImages []string) error {
...
+				log.Error(err, "error during prune", "digest", digest, "name", idToTagListMap[digest])
 				continue
 			}
-
-			if err := c.deleteImage(backgroundContext, img); err != nil {
-				log.Error(err, "Error during prune", "image", img)
-				continue
-			}
-			log.Info("Prune successful", "image", img)
+			log.Info("prune successful", "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]: can we align this with "removed image" and just say prune successful at the end if it's success?

File: pkg/eraser/helpers.go [hunk 10]
```diff
@@ -54,51 +54,49 @@ func removeImages(c Client, targetImages []string) error {
...
-			if _, running := runningImages[img]; running {
+			if util.IsExcluded(excluded, digest, idToTagListMap) {
+				log.Info("image is excluded", "digest", digest, "name", idToTagListMap[digest])
 				continue
 			}
 
-			if util.IsExcluded(excluded, img, idToTagListMap) {
-				log.Info("Image is excluded", "image", img)
+			if err := c.deleteImage(backgroundContext, digest); err != nil {
+				log.Error(err, "error during prune", "digest", digest, "name", idToTagListMap[digest])
```
Comment (sozercan) [inline pkg/eraser/helpers.go]: can we align this too? can we reflect if we encountered an error at the end?

**Review by sozercan — APPROVED** [review]

**Pull request merged**